lunes_cms/static/js/toggle_document_form_fields.js

```diff
@@ -22,76 +22,79 @@ $(document).ready(() => {
 function showDuplicates(data, parent) {
   removeDuplicationCheckMessage();
 
-  result = document.createElement("div");
-  result.setAttribute("id", "result");
-
-
   if (data["word"]) {
     // If there is a duplicated word, use orange background
-    result.style.backgroundColor = "#ffbb4a";
-    result.style.padding = "25px"
+    result = createSearchResultField("#ffbb4a");
 
     // Show alert message
-    messageBox = document.createElement("div");
-    message = document.createTextNode(data["message"]);
-    messageBox.append(message);
-    result.append(messageBox);
+    addDetailInfo(result, "message", "normal");
+
     // Show the duplicated word with its word type
-    wordBox = document.createElement("div");
-    word = document.createTextNode(data["word"]);
-    wordBox.style.fontWeight = "bold";
-    wordBox.append(word);
-    result.append(wordBox);
+    addDetailInfo(result, "word", "bold");
+
     // Show its definition too for more detail
-    definitionBox = document.createElement("div");
-    definition = document.createTextNode(data["definition"]);
-    definitionBox.append(definition);
-    result.append(definitionBox);
+    addDetailInfo(result, "definition", "normal");
+
     // Show related training sets
-    trainingSetsBox = document.createElement("div");
-    trainingSets = document.createTextNode(data["training_sets"]);
-    trainingSetsBox.append(trainingSets);
-    result.append(trainingSetsBox);
+    addDetailInfo(result, "training_sets", "normal");
   } else {
     // If there is no duplicate, use green background
-    result.style.backgroundColor = "#72f399";
-    result.style.padding = "25px"
+    result = createSearchResultField("#72f399");
+
     // Show message that no duplicate was found
-    messageBox = document.createElement("div");
-    message = document.createTextNode(data["message"]);
-    messageBox.append(message);
-    result.append(messageBox);
+    addDetailInfo(result, "message", "normal");
   }
   
   parent.prepend(result);
 }
 
+// Create a zone in which result duplicate search will be shown
+function createSearchResultField(backgroundColor){
+  messageField = document.createElement("div");
+  messageField.setAttribute("id", "result");
+  messageField.style.padding = "25px";
+  messageField.style.backgroundColor = backgroundColor;
 
-$(document).ready(() => {
-  $("#id_word").change((event) => {
-    if ($(event.target).val().length > 0) {
-      $.ajax({
-        type: 'GET',
-        url: '/api/search_duplicate/' + $(event.target).val(),
-        dataType: "json",
-        success: function(data) {
-            showDuplicates(data, $(event.target).closest(".card-body"));
-        }
-      })
-    } else {
-      removeDuplicationCheckMessage();
-    }
-  });
-  $("#id_word_type").trigger("change");
-});
\ No newline at end of file
+  return result;
+}
+
+// Create and add detail information
+function addDetailInfo(resultField, infoType, fontWeight) {
```

Comment:
Nice, that's helpful! I think if we add more style values we should instead introduce an object that contains the properties and values we want to change, but for now I don't think this is necessary :thinking:

Reply:
I thought the same 🤔 Bad is that we don't know yet how it will develop/change in future. I'll leave it for now.